tools/validators/instance_validator/validate/instance_parser.py

```diff
@@ -101,31 +124,31 @@ def parse_yaml(filename):
     Returns the parsed YAML data in a strictyaml-provided datastructure
     which is similar to a Python dictionary.
   """
-  yaml = _load_yaml_with_schema(filename, _SCHEMA)
-
-  top_name = yaml.keys()[0]
-
-  if _TRANSLATION in yaml[top_name].keys():
-    translation = yaml[top_name][_TRANSLATION]
-    translation.revalidate(_TRANSLATION_SCHEMA)
-
-    # if translation is not UDMI compliant
-    if isinstance(translation.data, str):
-      if translation.data != _COMPLIANT:
-        print('Translation compliance improperly defined')
-        sys.exit(0)
-    else:
-      translation_keys = translation.keys()
-
-      for key in translation_keys:
-        try:
-          translation[key].revalidate(_TRANSLATION_DATA_SCHEMA)
-        except (ruamel.yaml.parser.ParserError,
-                syaml.exceptions.YAMLValidationError,
-                syaml.exceptions.DuplicateKeysDisallowed,
-                syaml.exceptions.InconsistentIndentationDisallowed,
-                ruamel.yaml.scanner.ScannerError) as exception:
-          print(exception)
-          sys.exit(0)
-
-  return yaml
+  print('Loading yaml file to validate with schema')
+  entity_instance_block = ''
+  block_ready_for_validation = False
+  all_content = {}
+  with open(filename) as file:
+    for line in file:
+      if _ENTITY_INSTANCE_PATTERN.match(line):
+        # wait until there is a one entity instance block
+        if block_ready_for_validation:
+          validated = _validate_entity_with_schema(entity_instance_block,
+                                                   _SCHEMA)
+          all_content.update(validated.data)
+          entity_instance_block = line
+        else:
+          if not _IGNORE_PATTERN.match(line):
+            entity_instance_block = entity_instance_block + line
+            block_ready_for_validation = True
+      else:
+        if not _IGNORE_PATTERN.match(line):
+          entity_instance_block = entity_instance_block + line
+          block_ready_for_validation = True
```

Comment:
this logic could be simpler and more configurable:

```
all_content = {}
found_entities=0
entity_instance_block = ''
with open(filename) as file:
  for line in file:
    if _ENTITY_INSTANCE_PATTERN.match(line):
      if found_entities == ENTITIES_PER_BATCH:
        _validateBlock(entity_instance_block, all_content)
        found_entities = 0
        entity_instance_block = ''
      else:
        found_entities += 1
    entity_instance_block = entity_instance_block + line
  if found_entities > 0:
        _validateBlock(entity_instance_block, all_content)
return all_content
      
_validateBlock(unvalidated_block, validated_blocks):
  validated = _validate_entity_with_schema(unvalidated_block, _SCHEMA)
  validated_blocks.update(validated.data)
```

Reply:
did some tests, the fastest is to keep it per entity.
Do you still want it to be configurable?